tests/e2e/nebulacluster_test.go

```diff
@@ -180,53 +190,228 @@ func TestNebulaCluster(t *testing.T) {
 			)
 		}
 
-		for upgradeCaseIdx := range tc.UpgradeCases {
-			upgradeCase := tc.UpgradeCases[upgradeCaseIdx]
-			upgradeFunc := upgradeCase.UpgradeFunc
-			if upgradeFunc == nil {
-				upgradeFunc = func(ctx context.Context, t *testing.T, cfg *envconf.Config) context.Context {
-					klog.V(4).InfoS("Upgrade NebulaCluster", "namespace", namespace, "name", name)
+		if tc.UpgradeCases != nil {
```

Comment:
Why need to add `if`?

Reply:
Thought I need to add `if` to avoid an exception because `UpgradeCases` is set to `nil` in the backup test cases, but it actually works without the `if`. Removed.